examples/molnet/train_molnet.py

```diff
@@ -103,135 +237,109 @@ def main():
-        predictor = GraphConvPredictor(
-            WeaveNet(weave_channels=weave_channels, hidden_dim=n_unit,
-                     n_sub_layer=n_sub_layer, n_atom=n_atom),
-            MLP(out_dim=class_num, hidden_dim=n_unit))
-    elif method == 'rsgcn':
-        print('Train RSGCN model...')
-        predictor = GraphConvPredictor(
-            RSGCN(out_dim=n_unit, hidden_dim=n_unit, n_layers=conv_layers),
-            MLP(out_dim=class_num, hidden_dim=n_unit))
+    # Load the train and validation parts of the dataset.
+    filenames = [dataset_part_filename(p, num_data)
+                 for p in ['train', 'valid']]
+
+    paths = [os.path.join(cache_dir, f) for f in filenames]
+    if all([os.path.exists(path) for path in paths]):
+        dataset_parts = []
+        for path in paths:
+            print('Loading cached dataset from {}.'.format(path))
+            dataset_parts.append(NumpyTupleDataset.load(path))
     else:
-        raise ValueError('[ERROR] Invalid method {}'.format(method))
+        dataset_parts = download_entire_dataset(dataset_name, num_data, labels,
+                                                method, cache_dir)
+    train, valid = dataset_parts[0], dataset_parts[1]
+
+#    # Scale the label values, if necessary.
+#    if args.scale == 'standardize':
+#        if task_type == 'regression':
+#            print('Applying standard scaling to the labels.')
+#            datasets, scaler = standardize_dataset_labels(datasets)
+#        else:
+#            print('Label scaling is not available for classification tasks.')
+#    else:
+#        print('No label scaling was selected.')
+#        scaler = None
 
+    # Set up the predictor.
+    predictor = set_up_predictor(method, n_unit, conv_layers, class_num)
+
+    # Set up the iterators.
     train_iter = iterators.SerialIterator(train, args.batchsize)
-    val_iter = iterators.SerialIterator(val, args.batchsize,
-                                        repeat=False, shuffle=False)
+    valid_iter = iterators.SerialIterator(valid, args.batchsize, repeat=False,
+                                          shuffle=False)
 
+    # Load metrics for the current dataset.
     metrics = molnet_default_config[dataset_name]['metrics']
     metrics_fun = {k: v for k, v in metrics.items()
                    if isinstance(v, types.FunctionType)}
-    # loss_fun = molnet_default_config[dataset_name]['loss']
-    task_type = molnet_default_config[dataset_name]['task_type']
+    loss_fun = molnet_default_config[dataset_name]['loss']
+
     if task_type == 'regression':
-        loss_fun = regression_loss_fun
-        model = Regressor(predictor, lossfun=loss_fun, metrics_fun=metrics_fun,
-                          device=args.gpu)
-        # TODO(nakago): Use standard scaler for regression task
+        model = Regressor(predictor, lossfun=loss_fun,
+                          metrics_fun=metrics_fun, device=args.gpu)
+        # TODO: Use standard scaler for regression task
```

Comment:
TODO(nakago): ?

Reply:
Sorry, I used an older script (as a template) which contained that line. I will remove the comment.

Reply:
It should be fixed now.